src/dreyfus_index_updater.erl

```diff
@@ -31,6 +31,7 @@ update(IndexPid, Index) ->
     erlang:put(io_priority, {view_update, DbName, IndexName}),
     {ok, Db} = couch_db:open_int(DbName, []),
     try
+        {ok, ExcludeIdRevs} = purge_index(Db, IndexPid, Index),
```

Comment:
@chguocloudant Yes, exactly. If a doc is partially purged, if you will get both a purge_seq change (where you can always just a delete doc from clouseau), and update_seq change where you can insert a new doc to clouseau.  If during purge_index, you just delete docs, the delete/update part will not run twice